tests/test-sessions/test-sessions-common/src/test/java/org/eclipse/jetty/server/session/DuplicateCookieTest.java

```diff
@@ -70,13 +73,19 @@ public void testMultipleSessionCookiesOnlyOneExists() throws Exception
             assertEquals(0, s4422.getRequests());
 
             //make a request with another session cookie in there that does not exist
+            CountDownLatch latch = new CountDownLatch(1);
+            scopeListener.setExitSynchronizer(latch);
             Request request = client.newRequest("http://localhost:" + port1 + contextPath + servletMapping + "?action=check");
             request.headers(headers -> headers.add("Cookie", "JSESSIONID=123")); //doesn't exist
             request.headers(headers -> headers.add("Cookie", "JSESSIONID=4422")); //does exist
             ContentResponse response = request.send();
             assertEquals(HttpServletResponse.SC_OK, response.getStatus());
             assertEquals("4422", response.getContentAsString());
 
+            //ensure request has finished processing so session will be completed
+            latch.await(5, TimeUnit.SECONDS);
```

Comment:
I think you should do `assertTrue(latch.await(5, TimeUnit.SECONDS))`, otherwise if it does time out it will return false and not throw. 

Same for the usages below.

Reply:
Done.